lib/white_bread/event_manager.ex

```diff
@@ -0,0 +1,36 @@
+defmodule WhiteBread.EventManager do
+  @moduledoc false
+
+  ## An Event Manager inspired by that in ExUnit but adapted for
+  ## WhiteBread.
+
+  @timeout 30_000
+
+  def start_link() do
+    import Supervisor.Spec
+    child = worker(GenServer, [], restart: :temporary)
+    Supervisor.start_link([child], strategy: :simple_one_for_one, name: __MODULE__)
+  end
+
+  def stop() do
+    for {_, pid, _, _} <- Supervisor.which_children(__MODULE__) do
+      GenServer.stop(pid, :normal, @timeout)
+    end
+    Supervisor.stop(__MODULE__)
```

Comment:
As you're using `Supervisor.stop/1` there's no need to individually stop the children. As they're linked they will be stopped along with the supervisor.